src/Angor/Client/Shared/FeeSelector.razor

```diff
@@ -0,0 +1,191 @@
+                <div class="d-flex flex-column align-items-center">
+                    <span class="fw-semibold">Priority</span>
+                    <small class="opacity-75">Faster & Premium</small>
+                </div>
+            </label>
+        </div>
+
+        <div class="custom-fee-input">
+            <div class="form-check mb-2">
+                <input class="form-check-input" type="checkbox" id="useCustomFee" 
+                       checked="@UseCustomFee" 
+                       @onchange="HandleCustomFeeToggle" />
+                <label class="form-check-label" for="useCustomFee">
+                    Use custom fee rate
+                </label>
+            </div>
+            
+            @if (UseCustomFee)
+            {
+                <div class="d-flex align-items-center">
+                    <input type="number" 
+                           class="form-control" 
+                           placeholder="Enter fee rate in sats/byte" 
+                           @bind="CustomFeeRate"
+                           @bind:event="oninput" 
+                           @onblur="CustomFeeRateChanged"
+                           min="1" />
+                    <span class="px-1">sats</span>
+                </div>
+                @if (CustomFeeRate < RecommendedMinFee)
+                {
+                    <small class="text-warning mt-1">
+                        <i>Warning: Low fees may cause delayed confirmation</i>
+                    </small>
+                }
+            }
+        </div>
+    </div>
+</div>
+
+@code {
+    [Parameter]
+    public decimal SendFee { get; set; }
+
+    [Parameter]
+    public string CoinTicker { get; set; } = string.Empty;
+
+    [Parameter]
+    public long FeeRate { get; set; }
+
+    [Parameter]
+    public int FeeBlockCount { get; set; }
+
+    [Parameter]
+    public int FeePosition { get; set; } = 2; // Default to Standard
+
+    [Parameter]
+    public int FeeMin { get; set; } = 1;
+
+    [Parameter]
+    public int FeeMax { get; set; } = 3;
+
+    [Parameter]
+    public long RecommendedMinFee { get; set; } = 5;
+
+    [Parameter]
+    public EventCallback<int> OnFeePositionChanged { get; set; }
+
+    [Parameter]
+    public EventCallback<long> OnCustomFeeRateChanged { get; set; }
+
+    private bool UseCustomFee { get; set; }
+    private long CustomFeeRate { get; set; } = 10;
+
+    protected override void OnInitialized()
+    {
+        base.OnInitialized();
+        
+        if (FeePosition < FeeMin || FeePosition > FeeMax)
+        {
+            FeePosition = 2; // Default to Standard if invalid
+        }
+    }
+
+    private async Task SetFeePosition(int position)
+    {
+        try
+        {
+            if (position >= FeeMin && position <= FeeMax)
+            {
+                FeePosition = position;
+                await OnFeePositionChanged.InvokeAsync(position);
+
+                if (UseCustomFee)
```

Comment:
we invoke the `OnFeePositionChanged` and then we invoke the `OnCustomFeeRateChanged`?
this will cause the code to sign the trx twice it seems to me